build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:build_runner/build_runner.dart';
+import 'package:build_config/build_config.dart';
+
+const _scriptDir = '.dart_tool/build';
+const scriptLocation = '$_scriptDir/build.dart';
+
+Future<Null> ensureBuildScript() async {
+  var scriptFile = new File(scriptLocation);
+  // TODO - how can we invalidate this?
+  if (scriptFile.existsSync()) return;
+  var scriptDir = new Directory(_scriptDir);
```

Comment:
nit: you can replace this line and the next with `scriptFile.createSync(recursive: true)`.

Reply:
Done